lib/components/Example-wrapper.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react"
+import { CodeBlock, anOldHope } from "react-code-blocks"
+import jsxToString from "react-element-to-jsx-string"
+import { ChevronUp, ChevronDown } from "lucide-react"
+
+type FixtureWrapperProps = {
+  children: React.ReactNode
+}
+
+export function ExampleWrapper({ children }: FixtureWrapperProps) {
+  let codeString = jsxToString(children)
+
+  // Handle nested HOCs and format the code string
+  codeString = codeString
+    .replace(/<\w+\([^<>]*?(\w+)Base\)([^>]*)>/g, "<$1$2>")
+    .replace(/<\w+\([^<>]*?(\w+)Base\)([^>]*)?\/>/g, "<$1$2/>")
+    .replace(/<(\w+)\)\s*([^>]*)>/g, "<$1 $2>")
+    .replace(/<\/\w+(?:\(\w+)*\((\w+)Base\)\)>/g, "</$1>")
+    .replace(/<(\w+)\s*\)\s*([^>]*)>/g, "<$1 $2>")
+
+  // Format array and object props as single lines
+  codeString = codeString
+    .replace(
+      /(\w+)=\{\s*\[(.*?)\]\s*\}/gs,
+      (match, propName, inner) =>
+        `${propName}={[${inner.replace(/\s+/g, " ").trim()}]}`,
+    )
+    .replace(
+      /(\w+)=\{\s*\{\s*(.*?)\s*\}\s*\}/gs,
+      (match, propName, inner) =>
+        `${propName}={{${inner.replace(/\s+/g, " ").trim()}}}`,
+    )
```

Comment:
I looked at cosmos documentation, it is possible to load in the code. It is not acceptable to adapt JSX to convert it to code.

https://reactcosmos.org/docs/plugins/server-plugins

You can also create a server middleware in our vite config (this might be easier if this project uses vite because vite has ways of serving static directories)

You can make a fetch request in this file to request the static file content